### PR TITLE
Fix agent-to-controller security killswitch UI

### PR DESCRIPTION
**Pull request by daniel-beck**
This is a regression introduced by #5425 in 2.307: The form field name was changed but no corresponding getter was added, so it's always disabled. When submitting the form without changing anything, agent-to-controller access control is disabled (and the admin monitor shows up).

### Proposed changelog entries

* The checkbox labeled "Enable Agent → Controller Access Control" in the form "Configure Global Security" would always start out as disabled. Submitting the form without checking it would then cause a configuration change (regression in 2.307).

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by Wadeck — APPROVED** [review]
Not tested but relatively frequent regression, thanks for the catch Daniel :)

File: core/src/main/java/jenkins/security/s2m/MasterKillSwitchConfiguration.java [hunk 1]
```diff
@@ -29,7 +29,15 @@
         return GlobalConfigurationCategory.get(GlobalConfigurationCategory.Security.class);
     }
 
+    @Deprecated
```
Comment (oleg-nenashev) [inline core/src/main/java/jenkins/security/s2m/MasterKillSwitchConfiguration.java:32]: Add Javadoc too?

**Review by oleg-nenashev — APPROVED** [review]
We may merge it in 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process

**Comment by oleg-nenashev** [comment]
> The form field name was changed but no corresponding getter was added, so it's always disabled

Sounds like something major to report. Maybe not critical though it is not fun for security concerns. No objections for handling it as major bug. Does not seem to require immediate out of order release, no reports from users at the moment

**Review by res0nance — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
@daniel-beck could you please update the changelog?

**Pull request merged**